SwiftCompilerSources/Sources/Optimizer/Utilities/ManglingUtils.swift

```diff
@@ -0,0 +1,32 @@
+//===--- ManglingUtils.swift --------------------------------------------==//
+//
+// This source file is part of the Swift.org open source project
+//
+// Copyright (c) 2014 - 2023 Apple Inc. and the Swift project authors
+// Licensed under Apache License v2.0 with Runtime Library Exception
+//
+// See https://swift.org/LICENSE.txt for license information
+// See https://swift.org/CONTRIBUTORS.txt for the list of Swift project authors
+//
+//===----------------------------------------------------------------------===//
+
+import OptimizerBridging
+import SIL
+
+/// The mangler for functions where arguments or effects are specialized.
+struct FunctionSignatureSpecializationMangler {
```

Comment:
I don't think we need a separate bridged struct for this. It should be possible to do this with a single function in `FunctionPassContext`.

Reply:
Changed